### ISSUE TITLE
Improve documents validation message

### ISSUE THREAD

**Issue by MattiSG** [OpenTermsArchive/engine#832]
Declaration validation does not consider an empty `documents` key as an error.

### What happened

I tried to validate with `npm run validate Example` this malformed declaration:

```json
{
  "name": "Example",
  "documents": { },
  "Terms of Service": {
    "fetch": "https://domain.example/tos",
    "select": "body"
  }
}
```

I got the following error:

```
data should not have additional elements
```

### What I expected

An error in the style of:

```
`documents` cannot be empty, it should contain at least one of “Terms of Service”, “Imprint”…
```

**Comment by martinratinaud** [comment]
I believe `documents` can be empty (at least for now)

We needed to do this for OkCupid so that we stop tracking it but still keep track that it ever existed here.
Let's discuss this in next planification then.

**Comment by MattiSG** [comment]
After discussion, the best option seems to be to load both history files and declaration files so that closed and suspended services are made available. However, this has important consequences. For example, the `name` key is currently declared only in the main service declaration, not in its history. Allowing history files to be first-class declarations means duplicating this information, with a risk of divergence.

Instead of allowing a double declaration, the best way to improve the error messages seems to reformat it in the testing code instead of displaying to the contributor the message provided by the validator. Indeed, the error is:

```json
{
  keyword: 'additionalProperties',
  dataPath: '',
  schemaPath: '#/additionalProperties',
  params: { additionalProperty: 'Privacy Policy' },
  message: 'should NOT have additional properties'
} 
```

…we could thus give additional details on how to fix this error.

**Comment by MattiSG** [comment]
@Manu1400 if you want to have a look, that could be a good way to improve the validator 🙂 @ndpnt could help you navigate the code if needed.

**Comment by shravan562** [comment]
can i work on this

**Comment by MattiSG** [comment]
@shravan562 of course, you're welcome to! In order to ensure a smooth implementation and review, before starting any work, please describe your implementation plan in this issue 🙂 Some knowledge of JSON-Schema validators will be important.

**Comment by MattiSG** [comment]
@DevvObiero hi Paul! Would you mind clarifying your thoughts on this issue? 🙂 We'd also be curious to understand how you landed on it.

**Comment by dharma3306** [comment]
Hi! I'm interested in working on this issue. If it's still available, I'd like to take it and submit a fix.

**Comment by MattiSG** [comment]
Considering how much spam contributions this issue generates, I remove the `good first issue` label to decrease its visibility.
